### PR TITLE
Update to Kafka 3.0.0

### LINKED ISSUE #126: Upgrade Kafka Streams to 3.0

### PR DESCRIPTION
**Pull request by philipp94831**
Fixes #126

### CODE REVIEW

**Review by yannick-roeder — APPROVED** [review]

File: src/test/java/com/bakdata/kafka/integration/StreamsCleanUpTest.java [hunk 1]
```diff
@@ -533,7 +529,10 @@ private KafkaStreamsApplication createComplexApplication() {
         application.setErrorTopic(errorTopicName);
         application.setBrokers(this.kafkaCluster.getBrokerList());
         application.setProductive(false);
-        application.setStreamsConfig(Map.of(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, "0"));
+        application.setStreamsConfig(Map.of(
+                StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, "0",
+                ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "10000"
```
Comment (VictorKuenstler) [inline src/test/java/com/bakdata/kafka/integration/StreamsCleanUpTest.java:534]: Why is this necessary now?

Comment (philipp94831) [reply src/test/java/com/bakdata/kafka/integration/StreamsCleanUpTest.java:534]: Because it was changed to 45s by default in Kafka 3.0. This then requires us to wait longer for clean up, thus further increasing runtime of the tests. Therefore, I just configured it to the old default (this was also the missing setting that broke most tests)

**Review by VictorKuenstler — APPROVED** [review]
